### PR TITLE
[SPARK-49988][BUILD] Remove unused Hadoop dependency management

### PR DESCRIPTION
**Pull request by pan3793**
### What changes were proposed in this pull request?

Remove unused vanilla hadoop dependency(and transitive deps) management, i.e. `hadoop-client`, `xerces:xercesImpl`, and inline deps defined in `hadoop3` because it's the only supported hadoop profile.

### Why are the changes needed?

Simplify pom.xml.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Pass CI and verified runtime jars are not affected by running`dev/test-dependencies.sh`.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

**Comment by pan3793** [comment]
cc @dongjoon-hyun @LuciferYang

File: pom.xml [hunk 1]
```diff
@@ -1544,16 +1458,6 @@
...
         <scope>test</scope>
       </dependency>
-      <!-- Managed up to match Hadoop in HADOOP-16530 -->
-      <!--
-        When upgrading `xercesImpl` version, also need to change
-        the version definition in `SparkBuild#DependencyOverrides`.
-      -->
-      <dependency>
-        <groupId>xerces</groupId>
-        <artifactId>xercesImpl</artifactId>
```
Comment (LuciferYang) [inline pom.xml:1554]: Is it because Apache Spark doesn't require the `xerces:xercesImpl` dependency?

Comment (LuciferYang) [reply pom.xml:1554]: If that's the case, should we clean up the corresponding binary LICENSE of `xerces:xercesImpl` ?

Comment (pan3793) [reply pom.xml:1554]: it's a transitive dep of hadoop vanilla client, spark does not directly use it.

Comment (pan3793) [reply pom.xml:1554]: right, updated LICENSE and NOTICE

**Review by LuciferYang — APPROVED** [review]
+1, LGTM(pending tests)

File: pom.xml (hunk 1 above)
Comment (LuciferYang) [reply pom.xml:1554]: It would be best to declare this change in the pr description.

Comment (pan3793) [reply pom.xml:1554]: updated

**Review by dongjoon-hyun — COMMENTED** [review]
Thank you. Is this tested with Maven too, @pan3793 ?

**Comment by pan3793** [comment]
@dongjoon-hyun I tested the maven build, run `dev/test-dependencies.sh` to ensure that runtime deps are affected, and just run UT of a few modules(sts, yarn) using maven locally.

**Comment by LuciferYang** [comment]
We can add the following code to `build_and_test.yml` to have GA execute a Maven test once.

```
maven-test:
    name: "Run Maven Test"
    permissions:
      packages: write
    uses: ./.github/workflows/maven_test.yml
```

**Comment by pan3793** [comment]
@LuciferYang thanks for the information, I trigger a Maven build and test on CI, https://github.com/pan3793/spark/actions/runs/11379852474, let's wait for the results.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @pan3793 and @LuciferYang .
Merged to master for Apache Spark 4.0.0 on February 2025.

**Pull request closed without merging**